jwst/ami/ami_analyze_step.py

```diff
@@ -70,28 +70,24 @@ def process(self, input):
         self.log.info(f'Initial rotation guess = {rotate} deg')
         self.log.info(f'Initial values to use for psf offset = {psf_offset}')
 
-        # Open the input data model. Can be 2D or 3D image, so use general DataModel
-        try:
-            input_model = datamodels.DataModel(input)
-        except ValueError as err:
-            raise RuntimeError(f"{err}. Input unable to be read into a DataModel.")
-
         # Make sure oversample is odd
         if oversample % 2 == 0:
             raise ValueError("Oversample value must be an odd integer.")
 
         if os.getenv('WEBBPSF_PATH') is None:
             raise RuntimeError("WEBBPSF_PATH environment variable is required for this step")
 
-        # Apply the LG+ methods to the data
-        oifitsmodel, oifitsmodel_multi, amilgmodel = ami_analyze.apply_LG_plus(input_model,
-                                           oversample, rotate,
-                                           psf_offset,
-                                           rotsearch_parameters,
-                                           src, bandpass, usebp, 
-                                           firstfew, chooseholes, affine2d,
-                                           run_bpfix
-                                           )
+        # Open the input data model. Can be 2D or 3D image
+        with datamodels.open(input) as input_model:
```

Comment:
This replaces the `DataModel` check with a few advantages, the main one being that the `with` usage will automatically close the input file when it finishes (or errors out).